tests/lib/rules/linebreak-style.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @fileoverview No mixed linebreaks
+ * @author Erik Mueller
+ * @copyright 2015 Erik Mueller. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var eslint = require("../../../lib/eslint"),
+    ESLintTester = require("eslint-tester");
+
+var EXPECTED_LF_MSG = "Expected linebreaks to be 'LF' but found 'CRLF'.",
+    EXPECTED_CRLF_MSG = "Expected linebreaks to be 'CRLF' but found 'LF'.";
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var eslintTester = new ESLintTester(eslint);
+eslintTester.addRuleTest("lib/rules/linebreak-style", {
+
+    valid: [
+        {
+            code: "var a = 'a',\n b = 'b';\n\n function foo(params) {\n /* do stuff */ \n }\n",
+            args: [2]
+        },
+        {
+            code: "var a = 'a',\n b = 'b';\n\n function foo(params) {\n /* do stuff */ \n }\n",
+            args: [2, "unix"]
+        },
+        {
+            code: "var a = 'a',\r\n b = 'b';\r\n\r\n function foo(params) {\r\n /* do stuff */ \r\n }\r\n",
+            args: [2, "windows"]
+        }
+    ],
+
+    invalid: [
+        {
+            code: "var a = 'a';\r\n",
+            args: [2],
+            errors: [{
+                line: 1,
+                col: 12,
+                message: EXPECTED_LF_MSG
+            }]
+        },
+        {
+            code: "var a = 'a';\r\n",
+            args: [2, "unix"],
+            errors: [{
+                line: 1,
+                col: 12,
+                message: EXPECTED_LF_MSG
+            }]
+        },
+        {
+            code: "var a = 'a';\n",
+            args: [2, "windows"],
+            errors: [{
+                line: 1,
+                col: 12,
+                message: EXPECTED_CRLF_MSG
+            }]
+        },
+        {
+            code: "var a = 'a',\n b = 'b';\n\n function foo(params) {\r\n /* do stuff */ \n }\r\n",
+            args: [2],
+            errors: [{
+                line: 4,
+                col: 23,
+                message: EXPECTED_LF_MSG
+            }]
+        },
+        {
+            code: "var a = 'a',\r\n b = 'b';\r\n\n function foo(params) {\r\n /* do stuff */ \n }\r\n",
```

Comment:
I think chances are if one is wrong, a lot more are wrong and people would use some sort of conversion tool to fix it rather than editing one at a time. So, I'm guessing flagging just the first instance is the most useful.